### PR TITLE
[feat] Add workflow-audit-emit-issues: file audit findings as grouped GitHub issues

### LINKED ISSUE #345: [feat] audit-codebase should offer to file findings as grouped GitHub issues
## Goal
After an audit, optionally turn findings into context-grouped GitHub issues in one step, instead of stopping at a rendered table + a throwaway remediation plan.

## Motivation
`workflow-codebase-audit` produces a ranked findings doc and (via the command's plan flow) a remediation plan file. Acting on findings today means manually re-typing each into an issue. This session surfaced the gap directly: the audit output had no native path to issues, so the findings were hand-carried into `capture`. An emit-to-issues step would make audits actionable by default.

## Surface
- [x] Slash command
- [x] Skill — `workflow-*`

## Sketch (optional)
A post-audit prompt or flag (e.g. `/swe-workbench:audit-codebase ... --emit issues`) that groups findings by subsystem/context and routes each group through the existing `capture` / `product-manager` issue-filing path — honoring template discovery, `--label`, and the preview-confirm gate.

## Acceptance (optional)
- [ ] Audit findings can be filed as GitHub issues without manual re-entry.
- [ ] Findings are grouped by subsystem/context (not one-per-finding, not one-big-issue).
- [ ] Filing honors the repo's issue templates and `--label` selection.
- [ ] A preview-confirm gate runs before any issue is created.

### PR DESCRIPTION
**Pull request by lugassawan**
## Summary
- Add `skills/workflow-audit-emit-issues/` — new orchestrator skill that groups codebase audit findings by subsystem (path prefix → label), discovers `.github/ISSUE_TEMPLATE/` + repo labels, renders a batch preview with `drop N`/`edit N` support, and files one `gh issue create --body-file` issue per subsystem on `confirm`.
- Add `## Step 4` to `commands/audit-codebase.md` — after the audit renders, offers "Found N findings across M subsystems. File them as context-grouped GitHub issues? (y/n)"; invokes `swe-workbench:workflow-audit-emit-issues` on accept.
- Register in `agents/shared/workflows.md`, `docs/catalog.md`, and `README.md`.
- Add `tests/test_workflow_audit_emit_issues_skill.py` (30 tests) and `tests/test_audit_codebase_command.py` (7 tests).

## Test Plan
- [x] Changed skills/commands/agents load without errors
- [x] Examples in the diff were exercised manually

### Type-tailored checks ([feat])
- [x] `pytest tests/test_workflow_audit_emit_issues_skill.py tests/test_audit_codebase_command.py -v` — all 30+7 pass
- [x] `pytest tests/ -q` — full suite green (1108 tests)
- [x] `bash scripts/validate.sh` — all checks pass (no cycles, catalog complete, line caps)
- [x] SKILL.md frontmatter: `orchestrator: true`, `name: workflow-audit-emit-issues`, description contains required keywords
- [x] SKILL.md line count ≤ 300 (currently 199)
- [x] No action-verb reference to `workflow-codebase-audit` in SKILL.md (cycle safety)
- [x] `audit-codebase.md` Step 4 contains `invoke` on same line as `workflow-audit-emit-issues`

Closes #345

### CODE REVIEW

File: skills/workflow-audit-emit-issues/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+| Reply | Action |
+|-------|--------|
+| `confirm` (literal) | Run each sidecar `gh issue create` line; return issue URLs. After all issues are filed successfully, delete the temp `.md` files and the `.cmd` sidecar. |
+| `drop N` | Remove group N from the batch; rewrite temp files and sidecar; re-print preview. |
+| `edit N` | Show group N's body; accept revised text; rewrite temp file; re-print preview. |
+| anything else | Re-prompt; do **not** file. |
+
+On `gh issue create` failure: surface the error and print the `.cmd` sidecar path for
+manual retry.
```
Comment (lugassawan) [inline skills/workflow-audit-emit-issues/SKILL.md]: **High — Partial-failure sidecar leaves retry hazardous**

Phase 4's failure path surfaces the original `.cmd` sidecar path for manual retry, but the sidecar still contains ALL `gh issue create` lines — including any that already succeeded. A user following the retry hint will duplicate already-filed issues. The skill's own "Re-running re-files duplicates" warning compounds this.

**Suggested fix:** after each successful `gh issue create` call, rewrite the sidecar to contain only the remaining (unfiled) lines. On failure, surface already-filed issue URLs plus the updated `.cmd` path:

```markdown
On `gh issue create` failure: surface the error, print the URLs of
successfully-filed issues, and rewrite the `.cmd` sidecar to contain
only the remaining (unfiled) lines. Print the updated sidecar path for manual retry.
```

File: tests/test_workflow_audit_emit_issues_skill.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        chain_refs = [r for r in refs if r in _AUDIT_CHAIN]
+        if not chain_refs:
+            continue
+        if _ACTION_RE.search(line) and not _POINTER_RE.search(line):
+            pytest_fail = ", ".join(f"`swe-workbench:{r}`" for r in chain_refs)
+            assert False, (
+                f"Line contains an action verb targeting audit-chain skill(s) {pytest_fail} — "
+                "this creates a dependency cycle. Use pointer words only (see, mirrors, etc.).\n"
+                f"  Line: {line.strip()}"
+            )
```
Comment (lugassawan) [inline tests/test_workflow_audit_emit_issues_skill.py:194]: **Medium — Cycle-guard test only covers `swe-workbench:`-prefixed refs; plain backtick refs are invisible**

`test_no_action_verb_to_audit_chain` only fires on lines matching  `swe-workbench:X` . SKILL.md uses plain refs throughout (e.g.  `workflow-codebase-audit` ), so a future edit adding  invoke `workflow-codebase-audit`  would pass the test while creating an unchecked back-edge.

**Suggested fix:** add a companion test covering plain backtick refs:

```python
_PLAIN_AUDIT_RE = re.compile(r'`(workflow-codebase-audit|workflow-bug-triage)`')

def test_no_action_verb_to_audit_chain_plain_refs():
    text = SKILL_MD.read_text()
    for line in text.splitlines():
        if line.lstrip().startswith('@'):
            continue
        if not _PLAIN_AUDIT_RE.search(line):
            continue
        if _ACTION_RE.search(line) and not _POINTER_RE.search(line):
            assert False, f'Plain ref with action verb — potential back-edge: {line.strip()}'
```

File: skills/workflow-audit-emit-issues/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+For each finding derive a **subsystem** from `file_line`:
+
+1. Take the directory portion of the path (strip the filename).
+2. Use up to the first 2 path segments as the grouping key.
+3. Display label = last segment of that key:
+   - `src/auth/handler.ts` → key `src/auth` → label **`auth`**
+   - `commands/audit-codebase.md` → key `commands` → label **`commands`**
+   - `scripts/validate.py` → key `scripts` → label **`scripts`**
+4. No path (tool-level or unknown finding) → subsystem **`misc`**.
```
Comment (lugassawan) [inline skills/workflow-audit-emit-issues/SKILL.md]: **Medium — Phase 1 grouping is silent on root-level files**

For a finding on a root-level file (e.g. `README.md:12`), the directory portion is empty. The spec covers "no path → `misc`" but not "empty directory → ?". An agent must guess between `misc` and an empty-string label.

**Suggested fix:** add an explicit rule:

```markdown
5. Root-level file (directory portion is empty, e.g., `README.md`) → subsystem **`root`**.
```

File: agents/shared/workflows.md [hunk 4]
```diff
@@ -4,6 +4,7 @@ All `swe-workbench` workflow and integration skills available in this plugin. Us
 
 - `swe-workbench:workflow-bug-triage` — Investigate-and-file-issue counterpart to /debug. Iron Law (no fix without root cause), 4-phase loop, files structured issue with code-path table and impact assessment.
 - `swe-workbench:workflow-cleanup-merged` — Post-merge cleanup: fast-forward main (which auto-cleans via rimba post-merge hook when active), then verify; falls back to `rimba remove` or `git worktree` shell path; deletes local + remote branch.
+- `swe-workbench:workflow-audit-emit-issues` — Post-audit filing: groups codebase audit findings by subsystem (path prefix), discovers issue templates and labels, renders a batch preview, and files one GitHub issue per subsystem on `confirm`. Counterpart to `workflow-codebase-audit` (read-only); filing mirrors `workflow-bug-triage` Phase 4.
```
Comment (lugassawan) [inline agents/shared/workflows.md]: **Low — New entry is out of alphabetical order**

The file was alphabetical before this PR. `workflow-audit-emit-issues` (a-u-d) should precede `workflow-bug-triage` (b-u-g). Cosmetic, but worth keeping consistent so grep-scanning the list is predictable.

File: tests/test_audit_codebase_command.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+def test_command_invokes_skill():
+    text = CMD_MD.read_text()
+    assert "invoke" in text.lower() and "workflow-audit-emit-issues" in text, (
+        "commands/audit-codebase.md must use an action verb to invoke workflow-audit-emit-issues"
+    )
+
+
+def test_command_invokes_skill_on_same_line():
+    text = CMD_MD.read_text()
+    matching = [
```
Comment (lugassawan) [inline tests/test_audit_codebase_command.py:39]: **Low — Ambiguous variable name `l` (E741)**

Single-char `l` is visually indistinguishable from `1` and `I`. Some lint configs flag this as E741.

```suggestion
        line for line in text.splitlines()
        if "invoke" in line.lower() and "workflow-audit-emit-issues" in line
```

File: skills/workflow-audit-emit-issues/SKILL.md (hunk 1 above)
Comment (lugassawan) [reply skills/workflow-audit-emit-issues/SKILL.md]: Addressed in 675dfae: partial-failure now rewrites sidecar to unfiled lines only and surfaces already-filed URLs before the retry path.

File: tests/test_workflow_audit_emit_issues_skill.py (hunk 2 above)
Comment (lugassawan) [reply tests/test_workflow_audit_emit_issues_skill.py:194]: Addressed in 675dfae: added test_no_action_verb_to_audit_chain_plain_refs covering plain backtick refs alongside the swe-workbench:-prefixed check.

File: skills/workflow-audit-emit-issues/SKILL.md (hunk 3 above)
Comment (lugassawan) [reply skills/workflow-audit-emit-issues/SKILL.md]: Addressed in 675dfae: added explicit rule 4 — root-level file (empty directory portion) maps to subsystem `root`.

File: agents/shared/workflows.md (hunk 4 above)
Comment (lugassawan) [reply agents/shared/workflows.md]: Addressed in 675dfae: moved workflow-audit-emit-issues above workflow-bug-triage to restore alphabetical order.

File: tests/test_audit_codebase_command.py (hunk 5 above)
Comment (lugassawan) [reply tests/test_audit_codebase_command.py:39]: Addressed in 675dfae: renamed l → line in test_command_invokes_skill_on_same_line.

File: tests/test_workflow_audit_emit_issues_skill.py [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    text = readme.read_text()
+    assert "workflow-audit-emit-issues" in text, (
+        "README.md Workflows bullet must contain 'workflow-audit-emit-issues'"
+    )
+
+
+# ── triggers.txt ─────────────────────────────────────────────────────────────
+
+def test_triggers_has_at_least_two_fixtures():
+    lines = [l.strip() for l in TRIGGERS.read_text().splitlines() if l.strip()]
```
Comment (lugassawan) [inline tests/test_workflow_audit_emit_issues_skill.py]: **Low — Surviving E741 ambiguous variable `l` in `test_triggers_has_at_least_two_fixtures`**

The prior Low fix renamed `l` → `line` in `test_audit_codebase_command.py`, but the same pattern exists here in the skill test file (original feature commit, not the fix commit). Linters will still flag E741.

```suggestion
    lines = [line.strip() for line in TRIGGERS.read_text().splitlines() if line.strip()]
```

Comment (lugassawan) [reply tests/test_workflow_audit_emit_issues_skill.py]: Addressed in d93ba10: renamed l → line in test_triggers_has_at_least_two_fixtures.
